localstack-core/localstack/services/kms/models.py

```diff
@@ -616,14 +641,27 @@ def _get_key_usage(self, request_key_usage: str, key_spec: str) -> str:
                 raise ValidationException(
                     "You must specify a KeyUsage value for all KMS keys except for symmetric encryption keys."
                 )
-            elif request_key_usage != "GENERATE_VERIFY_MAC":
+            elif request_key_usage != KeyUsageType.GENERATE_VERIFY_MAC:
                 raise ValidationException(
                     f"1 validation error detected: Value '{request_key_usage}' at 'keyUsage' "
                     f"failed to satisfy constraint: Member must satisfy enum value set: "
                     f"[ENCRYPT_DECRYPT, SIGN_VERIFY, GENERATE_VERIFY_MAC]"
                 )
             else:
-                return "GENERATE_VERIFY_MAC"
+                return KeyUsageType.GENERATE_VERIFY_MAC
+        elif request_key_usage == KeyUsageType.KEY_AGREEMENT:
+            if key_spec not in [
+                KeySpec.ECC_NIST_P256,
+                KeySpec.ECC_NIST_P384,
+                KeySpec.ECC_NIST_P521,
+                KeySpec.ECC_SECG_P256K1,
+                KeySpec.SM2,
+            ]:
+                raise ValidationException(
+                    f"KeyUsage {request_key_usage} is not compatible with KeySpec {key_spec}"
+                )
+            else:
+                return request_key_usage
```

Comment:
This `elif` block is in the same indentation as a check for the key spec. The other such `elif` blocks which check the key usage are inside the topmost `if` statement. 

It might be perfectly valid like this. I just want to point it out as a sanity check :)

Reply:
The key spec for the first part makes sure to validate the HMAC keys and the recently added part will verify only the `KEY_AGREEMENT` usage's key specifications. In either case they are mutually exclusive. Does this answer your ques?